quic/client/handshake/FizzClientContext.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#pragma once
+
+#include <quic/client/handshake/ClientHandshakeFactory.h>
+
+#include <fizz/client/FizzClientContext.h>
+#include <fizz/protocol/DefaultCertificateVerifier.h>
+
+namespace quic {
+
+class FizzClientHandshake;
+
+class FizzClientContext
+    : public ClientHandshakeFactory,
+      public std::enable_shared_from_this<FizzClientContext> {
+ public:
+  std::unique_ptr<ClientHandshake> makeClientHandshake(
+      QuicCryptoState& cryptoState) override;
+
+  const std::shared_ptr<const fizz::client::FizzClientContext>& getContext()
+      const {
+    return context_;
+  }
+
+  const std::shared_ptr<const fizz::CertificateVerifier>&
```

Comment:
for both these methods you dont need to return a reference right, you can return the shared_ptr directly.

Reply:
I wanted to avoid making a copies when not needed.